### ISSUE TITLE
binPkg not set correctly in 0.9.18 - interpeter cannot find binary package any more

### ISSUE THREAD

**Issue by michaelboke** [traefik/yaegi#1144]
The following program `sample.go` triggers a panic: on the `0.9.18` tag but works  on `0.9.17` tag.
Probably has something todo with the new fix for `package name extraction from import path` https://github.com/traefik/yaegi/commit/d16bd4bcdb8685f7ce88a0ee2261f7d945a6cc84

I can see a difference in the contents of `github.com/traefik/yaegi/interp.Exports.binPkg` between versions.
It looks like the last element of the package name is cutoff, therefore wrongly imported

0.9.17 
```
binPkg = {github.com/traefik/yaegi/interp.Exports}
	 0 =  ->
	 1 = github.com/mbict/bebe/dsl ->
	 2 = github.com/mbict/bebe/engine ->
```

0.9.18
```
binPkg = {github.com/traefik/yaegi/interp.Exports}
	 0 =  ->
	 1 = github.com/mbict/bebe ->
```

The example program
```go
package main

import (
	"github.com/traefik/yaegi/interp"
	"reflect"
)

var Symbols1 = map[string]map[string]reflect.Value{}
var Symbols2 = map[string]map[string]reflect.Value{}

func init() {
	Symbols1["github.com/mbict/bebe/dsl"] = map[string]reflect.Value{}
	Symbols2["github.com/mbict/bebe/engine"] = map[string]reflect.Value{}
}

func main() {
	i := interp.New(interp.Options{})

	i.Use(Symbols1)
	i.Use(Symbols2)

	script := `
package main

import (
	"github.com/mbict/bebe/dsl"
)
`
	_, err := i.Eval(script)
	if err != nil {
		panic(err)
	}
}
```

Expected result: v0.9.17
```console
$ go run ./sample.go
// output
Eval without error
```

Got: v0.9.18
```console
$ go run ./sample.go
// output
 panic: 5:2: import "github.com/mbict/bebe/dsl" error: unable to find source related to: "github.com/mbict/bebe/dsl". Either the GOPATH environment variable, or the Interpreter.Options.GoPath needs to be set
```

**Comment by michaelboke** [comment]
I found the cause of this in the `interp/interp.go` file, i put some comments on the function https://github.com/traefik/yaegi/commit/d16bd4bcdb8685f7ce88a0ee2261f7d945a6cc84#r52229954

But ny simply reverting this function to the old situation would fix this. But maybe someone or the author of the changes @mvertes  could give some context on the changes in that patch. Because that is not really clear what the intend of the fix was before i would even undo this with a MR.

**Comment by sryio** [comment]
I have this problem, too.

**Comment by mvertes** [comment]
TLDR: change the key from `github.com/mbict/bebe/dsl` to `github.com/mbict/bebe/dsl/dsl`

From version `v0.9.18`, the key format in `binPkg` has changed from `import_path` to `import_path/package_name`, where `import_path`, in your case, is `github.com/mbict/bebe/dsl` and package_name is `dsl`. 

In most cases the package name (`dsl`) is equal to the last element of `import_path`, but it is not mandatory, and we have to cope with the cases of for example `github.com/pelletier/go-toml` where  package name is `toml`, or `github.com/stripe/stripe-go/v72` where package name is `stripe`, etc.

You are affected by this change because you maintain manually your exported symbols (`extract` tool is updated). Sorry for the inconvenience.

**Comment by michaelboke** [comment]
Ahh thanks for the quick response. It is clear now and I do understand why.

Your proposed fix is indeed working. Maybe I should also start using the extract tool by now.

Thanks again, i'm closing this one
